client/lib/route/path.js

```diff
@@ -132,8 +133,8 @@ export function getStatsPathForTab( tab, siteIdOrSlug ) {
 		return getStatsDefaultSitePage( siteIdOrSlug );
 	}
 
-	if ( tab === 'insights' && ! siteIdOrSlug ) {
-		// Insights only supports single-site - Link to an overview for now
+	if ( ( tab === 'insights' || tab === 'googleMyBusiness' ) && ! siteIdOrSlug ) {
+		// Insights & GMB only supports single-site - Link to an overview for now
 		return getStatsDefaultSitePage();
 	}
```

Comment:
For the records (it took me a bit of time to figure this out), this code path is only triggered when you:

1. Log in to a WordPrees.com account with multiple sites
2. Open on http://calypso.localhost:3000/google-my-business/stats/:site
3. Click the `Switch Site` link in the sidebar
4. Click the `All My Sites` link in the sidebar
5. You should be redirected to the http://calypso.localhost:3000/stats/day

Right now it doesn't work since we are redirected to http://calypso.localhost:3000/google-my-business/stats/select-business-type. This is most probably related to the other routing issue I mentioned.

Reply:
We should probably hide the `Google My Business` tab when all sites are selected, similarly to the `Activity` tab:

<img width="625" alt="screenshot" src="https://user-images.githubusercontent.com/594356/38466556-373f1de4-3b2b-11e8-88fe-2405e363dd31.png">

Which disappears when clicking the `All My Sites` link:

<img width="623" alt="screenshot" src="https://user-images.githubusercontent.com/594356/38466565-61563504-3b2b-11e8-8a64-7f9938d55c7f.png">